src/gluonts/model/n_beats/_ensemble.py

```diff
@@ -0,0 +1,441 @@
+
+# First-party imports
+from gluonts.core.component import validated, from_hyperparameters
+from gluonts.core import fqname_for
+from gluonts.core.serde import dump_json, load_json
+from gluonts.dataset.common import Dataset
+from gluonts.dataset.field_names import FieldName
+from gluonts.dataset.loader import DataBatch
+from gluonts.model.estimator import Estimator
+from gluonts.model.forecast import Forecast, SampleForecast
+from gluonts.model.predictor import Predictor, RepresentableBlockPredictor
+from gluonts.core.exception import GluonTSHyperparametersError
+from gluonts.trainer import Trainer
+
+# Relative imports
+from ._network import VALID_LOSS_FUNCTIONS
+from ._estimator import NBEATSEstimator
+
+# None is also a valid parameter
+AGGREGATION_METHODS = "median", "mean", "none"
+
+
+class NBEATSEnsemblePredictor(Predictor):
+    """"
+    An ensemble predictor for N-BEATS.
+    Calling '.predict' will result in::
+
+        |predictors|x|dataset|
+
+    predictions, if aggregation_method is 'none', otherwise in::
+
+        |dataset|
+
+    Parameters
+    ----------
+    prediction_length
+        Prediction horizon.
+    freq
+        Frequency of the predicted data.
+    predictors
+        The list of 'RepresentableBlockPredictor' that the ensemble consists of.
+    aggregation_method
+        The method by which to aggregate the individual predictions of the models.
+        Either 'median', 'mean' or 'none', in which case no aggregation happens.
+        Default is 'median'.
+    """
+
+    def __init__(
+        self,
+        prediction_length: int,
+        freq: str,
+        predictors: List[RepresentableBlockPredictor],
+        aggregation_method: Optional[str] = "median",
+    ) -> None:
+        super().__init__(prediction_length, freq)
+
+        assert aggregation_method in AGGREGATION_METHODS
+
+        self.predictors = predictors
+        self.aggregation_method = aggregation_method
+
+    def set_aggregation_method(self, aggregation_method: str):
+        assert aggregation_method in AGGREGATION_METHODS
+        self.aggregation_method = aggregation_method
+
+    def serialize(self, path: Path) -> None:
+        # serialize some metadata
+        super().serialize(path)
+
+        # basically save each predictor in its own sub-folder
+        num_digits = len(str(len(self.predictors)))
+        for index, predictor in enumerate(self.predictors):
+            composite_path = path / f"predictor_{str(index).zfill(num_digits)}"
+            os.makedirs(str(composite_path))
+            predictor.serialize(composite_path)
+
+        # serialize all remaining constructor parameters
+        with (path / "parameters.json").open("w") as fp:
+            parameters = dict(
+                prediction_length=self.prediction_length,
+                freq=self.freq,
+                aggregation_method=self.aggregation_method,
+                num_predictors=len(self.predictors),
+            )
```

Comment:
Upon doing the evaluation I noticed that it would be useful to include some information here about what kind of network is serialised here. At this point there is basically no way to retrieve this information other than looking at the weights.

Reply:
Ok, I just missed the files, its already working, my bad.